### DISCUSSION TITLE
Error: Invalid credentials. Invalid username or password

### DISCUSSION THREAD

**Discussion by ComradeJay** [PrismarineJS/mineflayer#1933] [Q&A]
- [ ] The [FAQ](https://github.com/PrismarineJS/mineflayer/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Versions
 - mineflayer: 3.7.0
 - server: vanilla/spigot/paper 1.16.5
 - node: 14.16.1

## Detailed description of a problem
So the bot WORKS its fine if I use console and do ```node .``` it runs and works as intended however when I load it all into Heroku to host it, it fails… I get this error

2021-05-25T18:20:22.958640+00:00 app[worker.1]: starting discord bot...
2021-05-25T18:20:23.282956+00:00 app[worker.1]: Error: Invalid credentials. Invalid username or password.
2021-05-25T18:20:23.282964+00:00 app[worker.1]:     at call (/app/node_modules/yggdrasil/src/utils.js:35:40)
2021-05-25T18:20:23.282965+00:00 app[worker.1]:     at processTicksAndRejections (internal/process/task_queues.js:95:5)
2021-05-25T18:20:23.282965+00:00 app[worker.1]:     at async auth (/app/node_modules/yggdrasil/src/Client.js:18:12)
2021-05-25T18:20:23.683081+00:00 app[worker.1]: Activity set to ads in the lobby 
2021-05-25T18:23:57.877644+00:00 heroku[worker.1]: State changed from up to down
2021-05-25T18:23:59.027792+00:00 heroku[worker.1]: Stopping all processes with SIGTERM
2021-05-25T18:23:59.206368+00:00 heroku[worker.1]: Process exited with status 143

## What did you try yet?

I can't find anything like this, so I haven't been able to try anything else :/ 

## Your current code
```js
console.log("importing packages...")
const mineflayer = require('mineflayer')
const discord = require('discord.js')
const config = require('./config.json')

console.log("logging in...")
const client = new discord.Client()
const bot = mineflayer.createBot({
	host: "minehut.com",
	username: config.minecraft_username,
	password: config.minecraft_password
})

console.log("searching for pog pkers..")
bot.on('messagestr', (message, messagePosition, jsonMsg) => {
	if (message.startsWith("[AD]") && message.toLowerCase().includes("/join "+config.server_name)) {
		var msg = message.split("] ").pop().split(": /join")[0]
		client.channels.cache.get(config.discord_channel).send(message)
	}
})

console.log("starting error listener...")
bot.on('kicked', console.log)
bot.on('error', console.log)

console.log("starting discord bot...")
client.login(config.discord_token)
.then(() => {
  client.user.setActivity('ads in the lobby', { type: 'WATCHING' })
  .then(presence => console.log(`Activity set to ${presence.activities[0].name}`))
  .catch(console.error);
}) ```

## Expected behavior
What happens is it uses my account to login to lobbys and look for people doing /ad for my server then people do that it sends it into my discord server and another skripted bot reads it off into the MC server.

## Additional context
As I said it works when I use ```node .```, but Heroku doesn't seem to like that I have a username and password in the config file. The bot comes online still just doesn't do what it's meant to do,

**Comment by aesthetic0001** [comment]
can you try to put the credentials in the createbot instead of getting them from config

**Comment by ComradeJay** [comment]
> can you try to put the credentials in the createbot instead of getting them from config

C:\Users\Jay\Documents\GitHub\adverse\adverse.js:10
        username: Ashanti901,
                  ^

ReferenceError: Ashanti901 is not defined

i get that

**Comment by aesthetic0001** [comment]
should be string

**Comment by ComradeJay** [comment]
> should be string

I'm a noob how would thou format a string

**Comment by aesthetic0001** [comment]
quotes.

**Comment by aesthetic0001** [comment]
next time please ask in discussions.

**Comment by ComradeJay** [comment]
I still get the same error..

**Comment by aesthetic0001** [comment]
should be email for username field btw

**Comment by ComradeJay** [comment]
I still get the same error...

```jss
username: "CorrectEmail",
password: "CorrectPass"
```

**Reply by aesthetic0001** [reply]
i guess try to login with those credentials otherwise idk

**Reply by ComradeJay** [reply]
i am lol

**Reply by amoraschi** [reply]
Also if your account is a microsoft, add auth: 'microsoft'

**Reply by ComradeJay** [reply]
its a java account
